### DISCUSSION TITLE
Export chat to text (word or libreoffice) in the same format IPED shows the whatsapp

### DISCUSSION THREAD

**Discussion by Klarissa-ksj** [sepinf-inc/IPED#429] [Q&A]
We have an analyst that wants to copy a chat from IPED and paste it to Word because he wants to edit the text, highlight some parts.

The color of the chat in word shows the diference between who sended and who received the message, but this analyst would like the messages aligned to left or to right in word, equal to whatspp view.

Is it possible?

**Comment by lfcnassif** [comment]
The whatsapp chat reports generated are simply HTML files. He can double click the chat, save and open in MS Word/LibreOffice, but not sure if those softwares will open the html report with the original formatting.

He can also take a picture of the viewer (there is an item in options menu for that) and edit the picture in his favorite image editor.

**Reply by wladimirleite** [reply]
I had noticed this too. Both MS Word and LibreOffice don't render the generated HTML as nicely as the internal viewer or browsers. In particular all messages are aligned in the same way, making it harder to follow the conversation.
I don't see any easy workaround. 
Maybe some minor change in the generated HTML can improve the layout shown in Word/LibreOffice, while keeping the same format in the internal viewer.

**Reply by wladimirleite** [reply]
I think I found a couple of simple changes in the generated HTML that make it render in a better way when copied to (or inserted into) a MS Word / LibreOffice document.
It also seems that thumb images are not being rendered in MS Word.

An example of how it is displayed now:
![image](https://user-images.githubusercontent.com/7217001/113321208-bebc9f80-92e9-11eb-9faf-83a91220ee08.png)

After a couple of simple changes:
![image](https://user-images.githubusercontent.com/7217001/113321809-591ce300-92ea-11eb-93d8-6917cf75d882.png)

Although it is a very small enhancement, the modified version seems easier to follow.
@lfcnassif, do you think it is worth making this change (possibly moving this discussion to issues)?

**Reply by lfcnassif** [reply]
For sure! Thank you @tc-wleite for taking a look. Is this a copy and paste or have you opened the html report directly in Word/LibreOffice?

**Reply by lfcnassif** [reply]
I didn't find a button to move this to issues, maybe will need to open a new one.

**Reply by wladimirleite** [reply]
I tried opening the HTML and copying and pasting from IPED's viewer, and both results were the same.

**Reply by wladimirleite** [reply]
> I didn't find a button to move this to issues, maybe will need to open a new one.

Yes. I think GitHub users requested this feature, but currently there is no such option. 
I will create a new one.

**Comment by Klarissa-ksj** [comment]
Thank you. I'll wait for the changes! Great!
